### PR TITLE
Timetable styles: flexbox meetings and port of administrative to Jinja

### PR DESCRIPTION
**Pull request by pferreir**

### CODE REVIEW

File: indico/MaKaC/services/implementation/conference.py [hunk 1]
```diff
@@ -347,8 +348,7 @@ class ConferenceDefaultStyleModification(ConferenceTextModificationBase):
     Conference default style modification
     """
     def _handleSet(self):
-        styleMgr = HelperMaKaCInfo.getMaKaCInfoInstance().getStyleManager()
-        if self._value == styleMgr.getDefaultStyleForEventType(self._conf.getType()):
+        if self._value == theme_settings.defaults[self._conf.getType()]:
```
Comment (ThiefMaster) [inline indico/MaKaC/services/implementation/conference.py]: what about using the `Event.type` property instead of `getType`? that would also let you use `lecture` instead of `simple_event`

File: indico/MaKaC/webinterface/rh/conferenceDisplay.py [hunk 2]
```diff
@@ -101,26 +101,21 @@ def _process( self ):
         view = "static"
         wf = self.getWebFactory()
         if wf != None:
-            type = self.getWebFactory().getId()
+            evt_type = self.getWebFactory().getId()
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/rh/conferenceDisplay.py]: this seems the same logic used in `Conference.getType()` - any idea why this code doesn't simply call `self._conf.getType()`?

File: indico/core/db/sqlalchemy/locations.py [hunk 3]
```diff
@@ -202,16 +202,26 @@ def venue_name(self):
     def venue_name(self, venue_name):
         self.own_venue_name = venue_name
 
-    @property
-    def room_name(self):
-        """The name of the room where this item is located."""
+    def get_room_name(self, full=True):
+        """The name of the room where this item is located.
+
+            :param full: by default, if the room has a "friendly name" too
```
Comment (ThiefMaster) [inline indico/core/db/sqlalchemy/locations.py]: I don't think this part of the docstring is supposed to be indented

File: indico/htdocs/sass/themes/administrative.scss [hunk 4]
```diff
@@ -157,3 +169,21 @@ td.itemTopAlign{
 td.subItemOrder{
     font-family: Times;
 }
+
+.preformatted {
```
Comment (ThiefMaster) [inline indico/htdocs/sass/themes/administrative.scss]: wouldn't this make more sense in one of the global css files? having it in theme css files seems like a good way to miss it in some places

File: indico/htdocs/sass/themes/indico.scss [hunk 5]
```diff
@@ -17,9 +17,25 @@
...
+ $header-text-color: $light-gray !default;
+ $tt-primary-color: #3875a8 !default;
+ $tt-secondary-color: #05b3b3 !default;
+ $tt-secondary-pastel: #a8dede !default;
+
+ @mixin header-logo($image, $position: 15px, $padding-left: 150px, $scale: none) {
+     div.event-header {
+         background: $header-bg-color url($image) left top no-repeat;
+         background-position: $position;
+        background-size: $scale;
```
Comment (ThiefMaster) [inline indico/htdocs/sass/themes/indico.scss]: indentation

File: indico/modules/events/registration/__init__.py [hunk 6]
```diff
@@ -77,7 +75,7 @@ def _inject_event_header(event, **kwargs):
                 .order_by(db.func.lower(RegistrationForm.title))
                 .all())
 
-    registrations = get_published_registrations(event)
+    registrations = event.published_registrations
```
Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: you could access this directly inside the template

File: indico/modules/events/settings.py [hunk 7]
```diff
@@ -181,3 +184,28 @@ def delete_all(self, event):
...
+        self.defaults = settings['defaults']
+
+    @staticmethod
+    @memoize
+    def _load_settings(settings_file):
+        with open(settings_file) as f:
+            return yaml.load(f)
+
+    @memoize
+    def themes_for(self, event_type):
```
Comment (ThiefMaster) [inline indico/modules/events/settings.py]: `get_themes_for`?

File: indico/modules/events/templates/display/administrative/event.html [hunk 8]
```diff
@@ -0,0 +1,100 @@
+{% from 'events/display/indico/_common.html' import render_datetime %}
+{% from 'events/display/common/_manage_button.html' import render_manage_button %}
+{% from 'events/timetable/display/administrative/_common.html' import render_location %}
+
+{% macro render_event_time(event, timezone) %}
+    {% if event.start_dt.astimezone(timezone).date() == event.end_dt.astimezone(timezone).date() %}
+        {{ event.start_dt | format_date(format='full', timezone=timezone|string) }}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: is the `|string` on `timezone` actually needed?

File: indico/modules/events/templates/display/administrative/event.html [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        - <em class="time">{{ event.start_dt | format_time(format='medium', timezone=timezone) }}</em>
+    {% else %}
+        {% trans start_date=event.start_dt|format_date(format='full', timezone=timezone|string),
+                 start_time=event.start_dt|format_time(format='medium', timezone=timezone),
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: you `|string` the timezone for `format_date` but not `format_time`. if the functions are really that broken maybe it'd be better to fix them so they are both (not) broken and accept both a timezone string and a tzinfo (assuming that's what you want to convert from with the `|string`)

File: indico/modules/events/templates/display/administrative/event.html [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                 end_date=event.end_dt|format_date(format='full', timezone=timezone|string),
+                 end_time=event.end_dt|format_time(format='medium', timezone=timezone) %}
+            from {{start_date}} ({{start_time}}) to {{end_date}} ({{end_time}})
+        {% endtrans %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: `{% trans -%}...{%- endtrans %}` - otherwise there's awful surrounding whitespace in the extracted string

File: indico/modules/events/templates/display/administrative/event.html [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+<table class="eventWrapper">
+    <tr>
+        <td>
+            <div class="manage-button-container">
+                {{ render_manage_button(event, 'EVENT', toggle_notes=false, show_note_operations=true) }}
+            </div>
+            <div class="reportNumberHeader">
+                {% for ref in event.references %}
+                    {{ ref.value }}<br>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: do you actually want a `<br>` after the last entry?

File: indico/modules/events/templates/display/administrative/event.html [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+{% endmacro %}
+
+<table class="eventWrapper">
+    <tr>
+        <td>
+            <div class="manage-button-container">
+                {{ render_manage_button(event, 'EVENT', toggle_notes=false, show_note_operations=true) }}
+            </div>
+            <div class="reportNumberHeader">
+                {% for ref in event.references %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: sort by something?

File: indico/modules/events/templates/display/administrative/event.html [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+                <div class="eventHeader">
+                    {{ theme_settings.title_fr or
+                       'ORGANISATION EUROP&Eacute;ENNE POUR LA RECHERCHE NUCL&Eacute;AIRE' | safe }}
+                    <br>
+                    <span class="CERNTitle"> CERN </span>
+                    {{ theme_settings.title_en or
+                       'EUROPEAN ORGANIZATION FOR NUCLEAR RESEARCH' | safe }}
+                </div>
+                <div align="center">
+                    <hr width="50%"/>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: not xhtml ;)

File: indico/modules/events/templates/display/administrative/event.html [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+            </div>
+            {% block header %}
+                <br>&nbsp;<br>
+                <div class="eventHeader">
+                    {{ theme_settings.title_fr or
+                       'ORGANISATION EUROP&Eacute;ENNE POUR LA RECHERCHE NUCL&Eacute;AIRE' | safe }}
+                    <br>
+                    <span class="CERNTitle"> CERN </span>
+                    {{ theme_settings.title_en or
+                       'EUROPEAN ORGANIZATION FOR NUCLEAR RESEARCH' | safe }}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: no need for `|safe` here

File: indico/modules/events/templates/display/administrative/event.html [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+                {% endif %}
+                <br>
+            </div>
+            {% if event.description %}
+                <br/><br/>
+                <span class="textTitle">Description: </span>
+                <br/>
+                <span class="descriptionText">
+                    <span itemprop="description" class="event-description">
+                    </span>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: missing the actual description?

File: indico/modules/events/templates/display/cern_committees.html [hunk 16]
```diff
@@ -0,0 +1,18 @@
+{% extends 'events/display/administrative/event.html' %}
+
+{% block header %}
+    <table class="eventHeader">
+        <tr>
+            <td>
+                <div style="float:left;">
+                    <img src="{{ indico_config.config.ImagesBaseURL }}/images/smallcern.png">
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/cern_committees.html]: are you sure the `.config` is needed? if `indico_config` i the thing I added long time ago it should have a `__getattr__` that proxies e.g. `imagesBaseURL` to `getImagesBaseURL()`

File: indico/modules/events/themes.yaml [hunk 17]
```diff
@@ -0,0 +1,298 @@
+defaults:
+    conference: static
+    simple_event: lecture
+    lecture: lecture
+    meeting: standard
+definitions:
+  # Lectures - base
+  event:
+    event_types:
+    - simple_event
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: I'd move those to the previous line, e.g. `event_types: [simple_event]`, or at least indent them

File: indico/modules/events/timetable/templates/display/administrative/_common.html [hunk 18]
```diff
@@ -0,0 +1,61 @@
+{% macro render_attachments(item, exclude_document=false) %}
+    {% for folder in item.attached_items['folders'] %}
+        {% if not exclude_document or folder.title != 'document' %}
+            <a href="{{url_for('attachments.list_folder', folder, redirect_if_single=True)}}">
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]: `true`

File: indico/modules/events/timetable/templates/display/administrative/_common.html [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+                {{ folder.title }}
+            </a>
+        {% endif %}
+    {% endfor %}
+    {% for file in item.attached_items['files'] %}
+        <a href="{{ file.download_url }}">
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]: it doesn't work with `.files`?

File: indico/modules/events/timetable/templates/display/administrative/_common.html [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+            {{ file.title }}
+        </a>
+    {% endfor %}
+{% endmacro %}
+
+{% macro render_linked_attachments(item, link_documents=false) %}
+    {% if link_documents %}
+        {% for folder in item.attached_items.get('folders', []) %}
+            {% if folder.title == 'document' and  item.references %}
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]: extra space after `and`

File: indico/modules/events/timetable/templates/display/administrative/_contribution.html [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+        <td class="itemTopAlign actions">
+            {% if not print_mode %}
+                {{ render_manage_button(contrib, 'CONTRIBUTION', toggle_notes=false, show_notes=show_notes,
+                                        show_note_operations=true,
+                                        show_button=(not event.as_legacy.isClosed() and
+                                                     contrib.can_manage_attachments(session.user))) }}
+            {% endif %}
+        </td>
+        {% for subcontrib in contrib.subcontributions %}
+            {{ render_subcontribution(subcontrib, event, numeral=(numeral|string + '.' + loop.index|string),
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/administrative/_contribution.html]: `'%s.%s'|format(numeral, loop.index)`?

File: indico/modules/events/timetable/templates/display/indico/_common.html [hunk 22]
```diff
@@ -9,24 +9,20 @@
...
-                    {%- else -%}
-                        <span class="title">{{ reference.urn or reference.value }}</span>
-                    {%- endif %}
-                    </li>
-                {%- endfor -%}
-            </ul>
-        </td>
-    </tr>
+    <ul class="reference-list">
+        {%- for reference in item.references: -%}
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/indico/_common.html]: no `:`

File: indico/modules/events/timetable/templates/display/indico/_session_block.html [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+                    <span class="label">{{ ngettext("Convener", "Conveners", conveners|length) }}</span>:
+                    {{ render_users(conveners) -}}
+                </div>
+            {% endif %}
+
+
+            <table class="timetable-item-details session-details">
+                <tbody>
+                    {{ render_attachments(session) }}
+                    </tbody>
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/indico/_session_block.html]: indentation

File: indico/web/flask/blueprints/admin.py [hunk 24]
```diff
@@ -69,10 +69,6 @@
 admin.add_url_rule('/layout/social', 'adminLayout-saveSocial', admins.RHAdminLayoutSaveSocial, methods=('POST',))
 admin.add_url_rule('/layout/template-set', 'adminLayout-saveTemplateSet', admins.RHAdminLayoutSaveTemplateSet,
                    methods=('POST',))
-admin.add_url_rule('/layout/styles/timetable/', 'adminLayout-styles', admins.RHStyles, methods=('GET', 'POST'))
-admin.add_url_rule('/layout/styles/timetable/create', 'adminLayout-addStyle', admins.RHAddStyle,
-                   methods=('GET', 'POST'))
-admin.add_url_rule('/layout/styles/timetable/<templatefile>/delete', 'adminLayout-deleteStyle', admins.RHDeleteStyle)
 admin.add_url_rule('/layout/styles/conference/', 'adminConferenceStyles', admins.RHConferenceStyles)
```
Comment (ThiefMaster) [inline indico/web/flask/blueprints/admin.py]: is this one still used?

File: indico/MaKaC/webinterface/rh/conferenceDisplay.py (hunk 2 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/rh/conferenceDisplay.py]: No idea, but I'm changing that.

File: indico/core/db/sqlalchemy/locations.py (hunk 3 above)
Comment (pferreir) [inline indico/core/db/sqlalchemy/locations.py]: Indeed.

File: indico/htdocs/sass/themes/administrative.scss (hunk 4 above)
Comment (pferreir) [inline indico/htdocs/sass/themes/administrative.scss]: Yeah, maybe in `_event_display.scss`.

File: indico/modules/events/templates/display/administrative/event.html (hunk 8 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: Yes. We have to fix that :disappointed:

File: indico/modules/events/registration/__init__.py (hunk 6 above)
Comment (pferreir) [inline indico/modules/events/registration/__init__.py]: No, I couldn't. That's injected by the registration module, as a timetable hook.

File: indico/modules/events/settings.py (hunk 7 above)
Comment (pferreir) [inline indico/modules/events/settings.py]: Good point.

File: indico/modules/events/templates/display/administrative/event.html (hunk 9 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: Agreed. I'll look into that.

File: indico/modules/events/templates/display/administrative/event.html (hunk 10 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: Well spotted! Thanks!

File: indico/modules/events/templates/display/administrative/event.html (hunk 13 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: Erm.. copy-paste.

Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: I've actually removed most slashes from existing stuff, as I knew you'd bi... complain about it.

File: indico/modules/events/templates/display/administrative/event.html (hunk 14 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: `title_en` can have special chars in a different theme config. Or does `|` have precedence over `or`?

File: indico/modules/events/templates/display/administrative/event.html (hunk 15 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: Good point :smile:

File: indico/modules/events/templates/display/cern_committees.html (hunk 16 above)
Comment (pferreir) [inline indico/modules/events/templates/display/cern_committees.html]: Yeah, it's needed, apparently. Just tested it again.

File: indico/modules/events/themes.yaml (hunk 17 above)
Comment (pferreir) [inline indico/modules/events/themes.yaml]: great idea!

File: indico/modules/events/timetable/templates/display/administrative/_common.html (hunk 19 above)
Comment (pferreir) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]: What? Where?

File: indico/web/flask/blueprints/admin.py (hunk 24 above)
Comment (pferreir) [inline indico/web/flask/blueprints/admin.py]: Yep. Those are all CSS files available by default for conference customization. Quite pointless but used.

File: indico/modules/events/templates/display/cern_committees.html (hunk 16 above)
Comment (ThiefMaster) [inline indico/modules/events/templates/display/cern_committees.html]: This is very strange... `indico_config.config` is the legacy Config instance

```

In [6]: config.imagesBaseURL
Out[6]: '/images'

In [7]: config.config.ImagesBaseURL
AttributeError...
```

(`config` in the shell is the same as `indico_config` in jinja templates)

I believe it only works for you since Jinja converts the AttributeError to `Undefined` which shows up as an empty string.

File: indico/modules/events/registration/__init__.py (hunk 6 above)
Comment (ThiefMaster) [inline indico/modules/events/registration/__init__.py]: I meant in `events/registration/display/event_header.html` (you currently pass `event` to it, so in there you could access `event.published_registrations` directly)

File: indico/modules/events/templates/display/administrative/event.html (hunk 14 above)
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: I think it does. In any case, I would add parentheses to make it clear.

File: indico/modules/events/templates/display/administrative/event.html [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+                <br>
+                {% block location_time %}
+                    {% if event.has_location_info %}
+                        {{ render_location(event) }}
+                    {% endif %}
+                    {{ render_event_time(event, timezone=timezone) }}
+                {% endblock %}
+                <br>
+                {% block material %}
+                {% endblock %}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: i'd move that to the previous line so it's immediately clear that the block is empty

File: indico/modules/events/templates/display/compact.html [hunk 26]
```diff
@@ -0,0 +1,79 @@
+{% from 'events/display/common/_legacy.html' import render_event_time, render_location %}
+{% from 'events/display/common/_manage_button.html' import render_manage_button %}
+
+<table class= "eventHeader">
+<tr>
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/compact.html]: indentation

File: indico/modules/events/themes.yaml [hunk 27]
```diff
@@ -0,0 +1,248 @@
+defaults:
+    conference: static
+    simple_event: lecture
+    lecture: lecture
+    meeting: standard
+definitions:
+  # Lectures - base
+  event:
+    event_types: [simple_event]
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: could use `event.type` instead of `conf.getType()` in the places where you access this - that way you could use `lecture` instead of `simple_event`

File: indico/modules/events/themes.yaml [hunk 28]
```diff
@@ -0,0 +1,248 @@
...
+    stylesheet: indico.scss
+    template: indico/meeting.html
+    title: Indico style
+  standard_inline_minutes:
+    event_types: [meeting]
+    stylesheet: indico.scss
+    template: indico/meeting.html
+    title: Indico style - inline minutes
+    settings:
+        show_notes: true
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: too much indentation

File: indico/modules/events/themes.yaml [hunk 29]
```diff
@@ -0,0 +1,248 @@
...
+  indico-weeks-view:
+    event_types: [conference, meeting]
+    stylesheet: meetings/Weeks.css
+    template: weeks.html
+    title: Indico Weeks View
+
+  # Static
+  static:
+    event_types:
+    - conference [simple_event, meeting]
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: this looks wrong

File: indico/modules/events/themes.yaml [hunk 30]
```diff
@@ -0,0 +1,248 @@
...
+    stylesheet: egee.scss
+    template: indico/meeting.html
+    title: EGEE meeting
+  endotofpet:
+    event_types: [simple_event, meeting]
+    stylesheet: endotofpet.scss
+    template: indico/meeting.html
+    title: EndoTOFPET
+  openlab:
+      event_types: [meeting]
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: indentation

File: indico/modules/events/themes.yaml [hunk 31]
```diff
@@ -0,0 +1,248 @@
...
+    settings:
+        link_documents: false
+  cern_committees:
+    event_types: [meeting]
+    stylesheet: cern_committees.scss
+    template: cern_committees.html
+    tt_template: administrative/event.html
+    title: CERN Committees
+    settings:
+        session_attachments: true
```
Comment (ThiefMaster) [inline indico/modules/events/themes.yaml]: indentation (a few more below)

File: indico/modules/events/timetable/templates/display/administrative/_common.html (hunk 19 above)
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]:
```
{% for file in item.attached_items['files'] %}
{% for file in item.attached_items.files %}
```

File: migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py [hunk 32]
```diff
@@ -0,0 +1,49 @@
...
+# revision identifiers, used by Alembic.
+revision = '3d672dc5ee53'
+down_revision = '29232c09e58a'
+
+
+def _move_theme(from_, to):
+    op.execute("""
+        UPDATE events.settings
+        SET value = '"%s"'::json
+        WHERE name = 'timetable_theme'
```
Comment (ThiefMaster) [inline migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py]: I'd also add a `module = '...'` since the name by itself is not enough to uniquely identify a setting (even though it works since the name is unique in this case, it probably won't make use if the index we have)

File: indico/modules/events/timetable/templates/display/administrative/_common.html (hunk 19 above)
Comment (pferreir) [inline indico/modules/events/timetable/templates/display/administrative/_common.html]: Yeah, true.

File: indico/modules/events/themes.yaml (hunk 28 above)
Comment (pferreir) [inline indico/modules/events/themes.yaml]: My editor :rage:

File: indico/modules/events/themes.yaml (hunk 29 above)
Comment (pferreir) [inline indico/modules/events/themes.yaml]: Thanks for spotting!

File: migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py (hunk 32 above)
Comment (pferreir) [inline migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py]: Done!

File: indico/modules/events/templates/display/administrative/event.html (hunk 14 above)
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: Just tested it, you do need parentheses:

```
>>> render_template_string('{{ "foo" or "bar" | tojson }}')
u'foo'
>>> render_template_string('{{ ("foo" or "bar") | tojson }}')
u'"foo"'
```

However, would you actually use html entities in the settings instead of just using the unicode/UTF8 character? (FWIW, I would also not use the html entity but the actual character in the french title)

File: indico/core/db/sqlalchemy/locations.py [hunk 33]
```diff
@@ -202,16 +202,26 @@ def venue_name(self):
     def venue_name(self, venue_name):
         self.own_venue_name = venue_name
 
-    @property
-    def room_name(self):
-        """The name of the room where this item is located."""
+    def get_room_name(self, full=True):
```
Comment (ThiefMaster) [inline indico/core/db/sqlalchemy/locations.py]: @nop33 didn't i comment something on one of your PRs about using `room_name` vs `room.name`? I think once this is merged you could simply use `get_room_name(full=False)`

File: indico/modules/events/models/events.py [hunk 34]
```diff
@@ -322,6 +328,11 @@ def participation_regform(self):
         return self.registration_forms.filter_by(is_participation=True, is_deleted=False).first()
 
     @property
+    def published_registrations(self):
```
Comment (ThiefMaster) [inline indico/modules/events/models/events.py]: `@memoize_request` - you already call it multiple times in the same template and right now that sends multiple queries

File: indico/modules/events/requests/templates/event_request_details.html [hunk 35]
```diff
@@ -54,25 +54,18 @@
...
-            {% endif %}
-            {% block form %}
-                {{ form_rows(form) }}
-            {% endblock %}
-        {% endcall %}
-    {% else %}
-        {{ self.form() }}
-    {% endif %}
+    {% call form_fieldset(legend=_('Request details'), description=_('These are the details provided by the event manager.')) %}
+            {% call form_row_static(_('Created by')) %}
```
Comment (ThiefMaster) [inline indico/modules/events/requests/templates/event_request_details.html]: too much indentation

File: indico/modules/events/templates/display/administrative/event.html [hunk 36]
```diff
@@ -0,0 +1,89 @@
...
+                    {% if not loop.last %}
+                        <br>
+                    {% endif %}
+                {% endfor %}
+            </div>
+            {% block header %}
+                <br>&nbsp;<br>
+                <div class="eventHeader">
+                    {{ theme_settings.title_fr or
+                       'ORGANISATION EUROP&Eacute;ENNE POUR LA RECHERCHE NUCL&Eacute;AIRE' | safe }}
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: I'd use the actual unicode characters - using UTF8 in templates is fine. If you do the same thing in the settings you don't need `|safe` at all

File: indico/modules/events/templates/display/administrative/event.html (hunk 14 above)
Comment (ThiefMaster) [inline indico/modules/events/templates/display/administrative/event.html]: Just tested it, you do need parentheses:

```
>>> render_template_string('{{ "foo" or "bar" | tojson }}')
u'foo'
>>> render_template_string('{{ ("foo" or "bar") | tojson }}')
u'"foo"'
```

However, would you actually use html entities in the settings instead of just using the unicode/UTF8 character? (FWIW, I would also not use the html entity but the actual character in the french title)

File: indico/modules/events/templates/display/pension_fund.html [hunk 37]
```diff
@@ -0,0 +1,25 @@
+{% extends 'events/display/administrative/event.html' %}
+
+{% block header %}
+    <br>&nbsp;<br>
+    <table class="eventHeader">
+       <tr>
+           <td>
+               <img src="{{ indico_config.config.ImagesBaseURL }}/images/CERNLogo.jpg">
```
Comment (ThiefMaster) [inline indico/modules/events/templates/display/pension_fund.html]: another broken image url reference that wouldn't work with indico outside `/`

File: indico/modules/events/timetable/templates/display/_weeks.html [hunk 38]
```diff
@@ -0,0 +1,149 @@
...
+
+{% macro _render_tooltip(entry, timezone) %}
+    <div class="tooltip hidden">
+        <strong>{{ entry.title }}</strong>
+        <br>
+        {% set speakers = entry.object.person_links|selectattr("is_speaker")|list %}
+        {% if entry.type.name == 'CONTRIBUTION' and speakers %}
+             {{ render_users(speakers, span_class="compact-speakers", separator=' ', title=false) }}<br>
+        {% endif %}
+        {{ entry.start_dt|format_time(timezone=timezone) }} -
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/_weeks.html]: `x|y` in this line, `x | y` in the next line

File: indico/modules/events/timetable/templates/display/compact.html [hunk 39]
```diff
@@ -0,0 +1,175 @@
+{% from 'attachments/_display.html' import render_attachments_folders %}
+{% from 'events/display/common/_manage_button.html' import render_manage_button %}
+{% from 'events/display/indico/_common.html' import render_users %}
+
+
+{% set day_list = days|list %}
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/compact.html]: why is this needed? is `days` a rrule/generator? in that case i'd rather listify it outside the template (passing generators to templates is a bit weird)

File: indico/modules/events/timetable/templates/display/compact.html [hunk 40]
```diff
@@ -0,0 +1,175 @@
...
+
+{% macro render_entry(entry, timezone, index) %}
+    {% if entry.type.name == 'CONTRIBUTION' -%}
+        {{ render_contribution(entry.object, timezone, index) }}
+    {% elif entry.type.name == 'SESSION_BLOCK' -%}
+        {{ render_session_block(entry.object, timezone) }}
+    {% elif entry.type.name == 'BREAK' -%}
+        {{ render_break(entry.object, timezone) }}
+    {% endif %}
+
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/templates/display/compact.html]: is the blank line here intentional?

File: indico/modules/events/timetable/views/__init__.py [hunk 41]
```diff
@@ -16,21 +16,26 @@
...
+import posixpath
+from itertools import groupby
 from operator import attrgetter
 
 from flask import render_template, request, session
 from pytz import timezone
 from sqlalchemy.orm import joinedload
 
-from indico.modules.events.layout import layout_settings
+from .weeks import inject_week_timetable
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/views/__init__.py]: imho it's pretty ugly to mix absolute and relative imports in the same file

File: indico/modules/events/timetable/views/weeks.py [hunk 42]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def inject_week_timetable(event, days, tz_name):
+    first_week_day = request.args.get('first_week_day', 'monday')  # monday/sunday/event
+    sunday_first = (first_week_day == 'sunday')
+    show_end_times = request.args.get('showEndTimes') == '1'
+
+    tz = timezone(tz_name)
+
+    day_dict = defaultdict(list, [(d, i) for d, i in days])
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/views/weeks.py]: `defaultdict(list, days)` doesn't work?

File: indico/modules/events/timetable/views/weeks.py [hunk 43]
```diff
@@ -0,0 +1,91 @@
...
+    sorted_dates = list(d.date() for d in iterdays(event.start_dt.astimezone(tz), event.end_dt.astimezone(tz)))
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/views/weeks.py]: `sorted_dates = [...]`

File: migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py [hunk 44]
```diff
@@ -0,0 +1,50 @@
...
+Revises: 29232c09e58a
+Create Date: 2016-03-17 10:03:17.857070
+"""
+
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '3d672dc5ee53'
+down_revision = '29232c09e58a'
```
Comment (ThiefMaster) [inline migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py]: `506ad8bf647c`

File: migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py [hunk 45]
```diff
@@ -0,0 +1,50 @@
+"""Unify lecture style ids
+
+Revision ID: 3d672dc5ee53
+Revises: 29232c09e58a
```
Comment (ThiefMaster) [inline migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py]: `506ad8bf647c`

Comment (ThiefMaster) [inline migrations/versions/201603171003_3d672dc5ee53_unify_lecture_style_ids.py]: Also update the filename so it's in the right order (I'd simply go for whatever the current time is when you rebase and update this file)

File: indico/modules/events/registration/__init__.py (hunk 6 above)
Comment (pferreir) [inline indico/modules/events/registration/__init__.py]: Ah indeed. Sorry, misread that.

File: indico/modules/events/templates/display/administrative/event.html (hunk 14 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: I really avoid adding special chars in code/tpl files. You never know what's gonna happen and Python doesn't have a very good record when it comes to handling system encodings nicely :/

File: indico/modules/events/templates/display/cern_committees.html (hunk 16 above)
Comment (pferreir) [inline indico/modules/events/templates/display/cern_committees.html]: Ok, nevermind, I was being an iditot :full_moon_with_face:

Comment (ThiefMaster) [inline indico/modules/events/templates/display/cern_committees.html]: *idiot

scnr

File: indico/core/db/sqlalchemy/locations.py (hunk 33 above)
Comment (nop33) [inline indico/core/db/sqlalchemy/locations.py]: Yes you did. OK, I'll keep in mind.

Comment (nop33) [inline indico/core/db/sqlalchemy/locations.py]: Can't I cherry-pick it?

Comment (ThiefMaster) [inline indico/core/db/sqlalchemy/locations.py]: yeah

File: indico/modules/events/templates/display/administrative/event.html (hunk 36 above)
Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: I'm usually wary of using unicode chars in source code. It's true that Jinja can handle it fine, but it's a matter of keeping it ASCII-friendly.

Comment (pferreir) [inline indico/modules/events/templates/display/administrative/event.html]: But yeah, maybe it's better to go UTF-8 and remove that hack.

File: indico/modules/events/timetable/templates/display/compact.html (hunk 39 above)
Comment (pferreir) [inline indico/modules/events/timetable/templates/display/compact.html]: Indeed not sure how that happened :rage3:

File: indico/modules/events/timetable/templates/display/compact.html (hunk 40 above)
Comment (pferreir) [inline indico/modules/events/timetable/templates/display/compact.html]: Nope! Thanks!

File: indico/modules/events/timetable/views/__init__.py (hunk 41 above)
Comment (pferreir) [inline indico/modules/events/timetable/views/__init__.py]: OK, let's discuss this IRL or on IRC (`IR?`), it's pretty important to define a good practice.

File: indico/modules/events/timetable/views/weeks.py (hunk 42 above)
Comment (pferreir) [inline indico/modules/events/timetable/views/weeks.py]: Yeah, that used to be a list, then a generator... long story. Thanks for spotting.
